### PR TITLE
BKNDLSS-11252 added remove of system properties (user-token, user-reg…

### PR DESCRIPTION
**Pull request by AnatoliiStepaniuk**
…istered) for BackendlessUser when saving(updating)

### CODE REVIEW

File: src/com/backendless/Persistence.java [hunk 1]
```diff
@@ -854,4 +868,12 @@ public void callStoredProcedure( String procedureName, Map<String, Object> argum
     Object[] args = new Object[] { Backendless.getApplicationId(), Backendless.getVersion(), procedureName, arguments };
     Invoker.invokeAsync( PERSISTENCE_MANAGER_SERVER_ALIAS, "callStoredProcedure", args, responder );
   }
+
+  private static void removeUserSystemProperties(BackendlessUser user)
+  {
+      if( user.getProperties().containsKey( "user-token" ) )
```
Comment (markpiller) [inline src/com/backendless/Persistence.java]: the "if" statement is not necessary, just calling remove is sufficient (and would work faster)

Also, create constants for the string values

File: src/com/backendless/UserService.java [hunk 2]
```diff
@@ -779,4 +783,10 @@ private boolean isLogoutFaultAllowed( String errorCode )
   {
     return errorCode.equals( "3064" ) || errorCode.equals( "3091" ) || errorCode.equals( "3090" ) || errorCode.equals( "3023" );
   }
+
+    private static void removeUserSystemProperties(BackendlessUser user)
```
Comment (scadgek) [inline src/com/backendless/UserService.java]: add javadoc for this method

File: src/com/backendless/UserService.java [hunk 3]
```diff
@@ -38,6 +38,8 @@
 {
   final static String USER_MANAGER_SERVER_ALIAS = "com.backendless.services.users.UserService";
   private final static String PREFS_NAME = "backendless_pref";
+  public static final String SOCIAL_USER_TOKEN = "user-token";
```
Comment (scadgek) [inline src/com/backendless/UserService.java]: user-token is not social property, also consider reusing it from HeaersManager.HeadersEnum.USER_TOKEN_KEY

File: src/com/backendless/UserService.java [hunk 4]
```diff
@@ -779,4 +788,15 @@ private boolean isLogoutFaultAllowed( String errorCode )
   {
     return errorCode.equals( "3064" ) || errorCode.equals( "3091" ) || errorCode.equals( "3090" ) || errorCode.equals( "3023" );
   }
+
+  /**
+   * When BackendlessUser is logged with Social network the returned BackendlessUser entity may contain some system properties
+   * like "user-token" or "user-registered". If such entity is used for operations like save, register or update an error will occur.
+   * To avoid this, such system properties should be removed from the entity before these methods are invoked.
+   */
+    private static void removeUserSystemProperties(BackendlessUser user)
```
Comment (markpiller) [inline src/com/backendless/UserService.java]: there is no reason to duplicate that method in two places. You could create a static method in BackendlessUser and use that from everywhere.

**Pull request merged**